### PR TITLE
Remove TaskScheduler.Default from CLangAnalyzer and update tests

### PR DESCRIPTION
**Pull request by bigfluffycookie**

### CODE REVIEW

File: src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs [hunk 1]
```diff
@@ -285,6 +287,27 @@ public async Task TriggerAnalysisAsync_AnalysisFailsDueToInternalMessage_Notifie
             statusNotifier.VerifyNoOtherCalls();
         }
 
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs:290]: This is what I ended up with, I wasn't sure if I needed to test anything ontop of the callback order so please let me know

File: src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs [hunk 2]
```diff
@@ -309,16 +332,20 @@ private static Mock<IRequestFactoryAggregate> CreateRequestFactory(string filePa
             IAnalysisStatusNotifier statusNotifier = null,
             ICFamilyIssueConverterFactory issueConverterFactory = null,
             IRequestFactoryAggregate requestFactory = null,
-            ILogger logger = null)
+            ILogger logger = null,
+            IFileSystem fileSystem = null,
```
Comment (bigfluffycookie) [inline src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs:336]: I added in the filesystem here aswell as it makes sense for it to be there

File: src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs:290]: I think that's the only new thing we want to test here.

File: src/CFamily/Analysis/CLangAnalyzer.cs [hunk 3]
```diff
@@ -122,7 +126,7 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
             // For notes on VS threading, see https://github.com/microsoft/vs-threading/blob/master/doc/cookbook_vs.md
```
Comment (duncanp-sonar) [inline src/CFamily/Analysis/CLangAnalyzer.cs:126]: Minor: I would probably delete this comment: we're now using our own abstraction layer, so the cookbook isn't directly relevant.

**Review by duncanp-sonar — COMMENTED** [review]
Nice: just a couple of minor comments.

File: src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs [hunk 4]
```diff
@@ -285,6 +287,27 @@ public async Task TriggerAnalysisAsync_AnalysisFailsDueToInternalMessage_Notifie
             statusNotifier.VerifyNoOtherCalls();
         }
 
+        [TestMethod]
+        public async Task TriggerAnalysisAsync_ThreadHandling_CorrectCallbackOrder()
```
[the remark this answers is not in the document]
Comment (bigfluffycookie) [reply src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs:291]: Good point, I sometimes get stuck on the naming convention of "methodName_TestScenario_ExpectedResult" so sometimes I come up with strange names trying to match it

**Pull request merged**